src/Forms/FieldList.php

```diff
@@ -705,10 +673,8 @@ public function transform($trans)
 
     /**
      * Returns the root field set that this belongs to
-     *
-     * @return FieldList|FormField
      */
-    public function rootFieldList()
+    public function rootFieldList(): static|FormField
```

Comment:
```suggestion
    public function rootFieldList(): FieldList
```
`containerField` can only be a `CompositeField`, and its `rootFieldList()` method returns either its children (which is a `FieldList`) or calls `rootFieldList()` on its `containerFieldList` which can only be a `CompositeField`.

Note we can't use static here because this might return `FieldList` from some subclass, which would be an error if we use `static`.

Reply:
The method includes `return $this`, so a static return type is correct

> Note we can't use static here because this might return FieldList from some subclass, which would be an error if we use static.

I think you're getting confused with `self`

Reply:
Updated